src/test/scala/chiselTests/LazyCloneSpec.scala

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.stage.ChiselStage
+import chiselTests.ChiselFlatSpec
+
+class LazyCloneSpec extends ChiselFlatSpec {
+  behavior.of("LazyClone")
+
+  it should "lazily clone" in {
+    object Foo {
+      var count = 0L
+    }
+
+    class Foo extends Bundle {
+      val a = UInt(8.W)
+      Foo.count += 1
+    }
+
+    class MyModule extends RawModule {
+      val io = IO(Flipped(new Bundle {
+        val foo = Output(new Foo)
+        val bar = Input(new Foo)
+      }))
+    }
+    ChiselStage.elaborate(new MyModule)
+    println(s"copies: ${Foo.count}")
+    Foo.count should be < 12L
```

Comment:
```suggestion
    Foo.count should be(2)
```

Also, it would be good to have a test that sharing a single `Foo` for the `Input` and `Output` does result in cloning. 
```scala
class MyModule extends RawModule {
  val foo = new Foo
   val io = IO(Flipped(new Bundle {
     val foo = Output(foo)
     val bar = Input(foo)
   }))
 }
// This should have 3 Foos constructed
```

```scala
object Bar {
   var count = 0L
 }

 class Bar(x: UInt) extends Bundle {
   val a = x
   Bar += 1
 }
class MyModule extends RawModule {
   val io = IO(Flipped(new Bundle {
     val foo = Output(new Bar(UInt(8.W)))
     val bar = Input(new Bar(UInt(8.W)))
   }))
 }
// This should have 4 Bars constructed
```

Also a test that shows an extra clone when you pass a `Data` object to the `Bundle` that is used as a field (aka an externalRef).